javascript/src/test/services/scheduleAPI.test.js

```diff
@@ -0,0 +1,266 @@
+import React from "react";
+import { render, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Home from "main/pages/Home/Home";
+import {fetchcreateScheduleJSON, fetchdeleteScheduleJSON, fetchgetScheduleJSON, fetchgetSchedulesJSON} from "main/services/scheduleAPI";
+
+import {fetchWithToken} from "main/utils/fetch";
+jest.mock("main/utils/fetch");
+import { useAuth0 } from "@auth0/auth0-react";
+jest.mock("@auth0/auth0-react");
+
+describe("ScheduleAPI tests",  () => {
+  test("fetchCreateSchedule", async () => {
+
+    const sampleReturnValue = {
+        id: 1,
+        name: "main",
+        description: "test",
+        userId: "123",
+        quarter: "W20",
+    };
+
+    fetchWithToken.mockResolvedValue({
+        status: 200,
+        json: () => {
+          return sampleReturnValue;
+        },
+      });
+
+
+    const expectedFields = {
+        name: "main",
+        description: "test",
+        userId: "123",
+        quarter: "W20"
+    };
+
+    const onSuccess = jest.fn();
+    const onError = jest.fn();
+
+    const result = fetchcreateScheduleJSON(expectedFields,useAuth0,onSuccess,onError);
```

Comment:
I don't think the value we pass to this function (and the other functions in your other test cases) is `useAuth0`, it's a function that returns the token. Can we make this less misleading by passing in a mock function instead?

Reply:
Thanks for the input, I just made a revision to account for this!